### PR TITLE
[Dialogs] Added text color customization for Alerts

### PR DESCRIPTION
**Pull request by yarneo**
This is my first attempt on adding text color customization to the alert controller component (see #2046 for reference).

I wanted it to fit well with our themer and the UIAppearance proxy. Therefore there were a few extra steps needed to ensure that.

First, UILabel doesn't have UI_APPEARANCE_SELECTOR for its textColor. Therefore I subclassed the content view that holds both UILabels for our alert controller and created 2 UIColors with the needed selector. Reason I didn't subclass the UILabel or create a category for it is because I wanted a way to differentiate between the two UILabels as to add more flexibility in changing the color for both the title and the message and not only one color for both.

Talking to the designer, there wasn't a good fit in MDCColorScheme for the text coloring (as text color is usually meant to be in the grayscale range), hence I added two new properties called primaryTextColor and secondaryTextColor so then the text can be customized through the themer. The default for the text color is the same as our current default coloring.

### CODE REVIEW

File: components/Themes/src/MDCColorScheme.h [hunk 1]
```diff
@@ -54,6 +54,17 @@
...
+ The text color to be typically used as titles and subjects. This will currently be used as the color
+ of the alert controller title.
+ */
+@property(nonatomic, strong, nonnull, readonly) UIColor *primaryTextColor;
+/**
+ The text color to be typically used as the body or message. This will currently be used as the color
+ of the alert controller message.
+ */
+@property(nonatomic, strong, nonnull, readonly) UIColor *secondaryTextColor;
+
```
Comment (ianegordon) [inline components/Themes/src/MDCColorScheme.h]: Naming is hard.
To have primaryColor, primaryLightColor and primaryDarkColor be completely unrelated to primaryTextColor may be confusing.
Thoughts on alternative naming?  textColor, lightTextColor, darkTextColor?
Do we want to add backgroundColor?  Would that enable us to support Dark and Light theming through our schemes?
Naming is hard!

File: components/Dialogs/src/ColorThemer/MDCAlertColorThemer.m [hunk 2]
```diff
@@ -17,18 +17,20 @@
...
+      toAlertContentView:(MDCAlertContentView *)alertContentView {
   #if defined(__IPHONE_9_0) && __IPHONE_OS_VERSION_MIN_REQUIRED >= __IPHONE_9_0
   [[MDCButton appearanceWhenContainedInInstancesOfClasses:@[[MDCAlertController class]]]
       setTitleColor:colorScheme.primaryColor forState:UIControlStateNormal];
   #else
   [[MDCButton appearanceWhenContainedIn:[MDCAlertController class], nil]
       setTitleColor:colorScheme.primaryColor forState:UIControlStateNormal];
   #endif
+  [alertContentView setTitleLabelColor:colorScheme.primaryTextColor];
+  [alertContentView setMessageLabelColor:colorScheme.secondaryTextColor];
```
Comment (ianegordon) [inline components/Dialogs/src/ColorThemer/MDCAlertColorThemer.m]: Prefer property invocations on properties:
alertContentView.titleLabelColor = colorScheme.primaryTextColor
alertContentView.messageLabelColor = colorScheme.secondaryTextColor

File: components/Dialogs/src/MDCAlertContentView.h [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+ */
+
+#import <UIKit/UIKit.h>
+
+@interface MDCAlertContentView : UIScrollView
+
+@property(nonatomic, strong, nullable) UILabel *titleLabel;
```
Comment (ianegordon) [inline components/Dialogs/src/MDCAlertContentView.h:21]: readonly?

File: components/Dialogs/src/MDCAlertContentView.m [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+  }
+  return self;
+}
+
+- (void)commonMDCAlertContentViewInit {
+  self.backgroundColor = [UIColor whiteColor];
+  
+  self.titleLabel = [[UILabel alloc] initWithFrame:CGRectZero];
+  self.titleLabel.numberOfLines = 0;
+  self.titleLabel.textAlignment = NSTextAlignmentNatural;
```
Comment (ianegordon) [inline components/Dialogs/src/MDCAlertContentView.m:42]: Don't we need to add titleLabel and messageLabel as subviews?

File: components/Themes/examples/ThemerCustomSchemePickerController.m [hunk 5]
```diff
@@ -181,7 +181,8 @@ - (void)applyColorScheme:(id<MDCColorScheme>)colorScheme {
   // Apply color scheme to material design components using component themers.
   [MDCActivityIndicatorColorThemer applyColorScheme:colorScheme
                                 toActivityIndicator:[MDCActivityIndicator appearance]];
-  [MDCAlertColorThemer applyColorScheme:colorScheme];
+  [MDCAlertColorThemer applyColorScheme:colorScheme
+                     toAlertContentView:[MDCAlertContentView appearance]];
```
Comment (ianegordon) [inline components/Themes/examples/ThemerCustomSchemePickerController.m]: Allow the CustomTheme picker to set custom text colors so we can verify the themer works as expected?

**Review by ianegordon — CHANGES_REQUESTED** [review]

**Comment by ianegordon** [comment]
Let's discuss expanding MDCColorScheme with the wider team.  There were some concerns from Will that we should address.

File: components/Dialogs/src/MDCAlertContentView.h (hunk 3 above)
Comment (yarneo) [reply components/Dialogs/src/MDCAlertContentView.h:21]: titleLabel is being set in MDCAlertController

File: components/Dialogs/src/MDCAlertContentView.m (hunk 4 above)
Comment (yarneo) [reply components/Dialogs/src/MDCAlertContentView.m:42]: they are added in MDCAlertController (as previous behavior)

**Comment by yarneo** [comment]
I removed the Schemer and ColorThemer additions as that needs further discussion. The current behavior is that to be able to change the title color of the MDCAlertController you can set it by calling `MDCAlertContentView.appearance().titleLabelColor = .blue` or `MDCAlertContentView.appearance().messageLabelColor = .red` for the message label.

File: components/Dialogs/src/MDCAlertContentView.m (hunk 4 above)
Comment (ianegordon) [reply components/Dialogs/src/MDCAlertContentView.m:42]: I'm not sure i've seen this before. Typically I would expect the view that instantiates UIViews to add them as subviews.
Let's discuss when you are free.

File: components/Dialogs/src/MDCAlertContentView.m [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+ */
+
+#import "MDCAlertContentView.h"
+
+@implementation MDCAlertContentView
```
Comment (ianegordon) [inline components/Dialogs/src/MDCAlertContentView.m:19]: FYI: If this is a private implementation detail, we tend to create these files under src/private.  It's Objective-c, so we can't enforce any privacy, but hopefully it indicates to our clients that these classes / code are not intended to be directly used, subclasses, etc.

File: components/Dialogs/src/MDCAlertContentView.m [hunk 7]
```diff
@@ -0,0 +1,59 @@
...
+  
+  self.messageLabel = [[UILabel alloc] initWithFrame:CGRectZero];
+  self.messageLabel.numberOfLines = 0;
+  self.messageLabel.textAlignment = NSTextAlignmentNatural;
+}
+
+- (void)setTitleLabelColor:(UIColor *)titleLabelColor {
+  _titleLabelColor = titleLabelColor;
+  self.titleLabel.textColor = _titleLabelColor;
+}
```
Comment (ianegordon) [inline components/Dialogs/src/MDCAlertContentView.m:52]: Exactly what we were talking about. +1

**Review by ianegordon — COMMENTED** [review]
Hi @yarneo , I like the API exposing the properties, but we may be able to simplify the implementation. Maybe not, let's catch up.

**Comment by ianegordon** [comment]
This is related to #2046.

We will likely follow the template we are using for customized fonts.  The end result will be very close to this, although the colors and fonts properties will be exposed directly on MDCAlertController.

PR (Mk 2) to follow.

**Pull request closed without merging**